glue/egmde.launcher

```diff
@@ -4,10 +4,15 @@ set -e
 # For a classic snap this needs to be something sane
 export XDG_RUNTIME_DIR=/run/user/$(id -u)
 
-# If there's another Wayland server, ignore WAYLAND_DISPLAY
-if [ -n "$WAYLAND_DISPLAY" ] && [ -e "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY.lock" ]
+# If there's an existing Wayland server, we're likely on a Wayland desktop
+if [ -n "$WAYLAND_DISPLAY" ] && [ -e "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY" ]
 then
-  unset WAYLAND_DISPLAY
+  # select an uncontested WAYLAND_DISPLAY
+  display=10
```

Comment:
Why start at `wayland-10`? You have logic just below to find a free slot. Why not start looking at `wayland-0` like every other compositor? (personally I care because I will often launch things with `WAYLAND_DISPLAY=wayland-1` without checking the actual display created)

Reply:
Well, to workaround the current behavior I've specified WAYLAND_DISPLAY manually, so I've avoided low numbers that might already be chosen automatically and this just reflects that practice.

Once https://github.com/MirServer/mir/pull/1396 is cherry-picked and released this PR could be reverted and gives that behavior. So that's what we should do.